### DISCUSSION TITLE
Archive function improvment

### DISCUSSION THREAD

**Discussion by Brainfighter** [crunchy-labs/crunchy-cli#302] [Ideas]
I currently use this command to download multiple shows:

crunchy-cli -q --user-agent "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 5.1; Trident/4.0)" archive -a en-US -a de-DE -a ja-JP -s en-US -s de-DE --skip-existing --yes --include-fonts -o "/mnt/loader/{series_name}/{series_name} - S{season_number}E{episode_number}.mkv" $(cat series.txt)

The problem I run into is that it will also download episodes that are only available in ja-JP. I run my command on a regular basis so that I get the updated and new episodes automatically. 

Would it be possible for the archive function to not only check if the file is already downloaded but also check the audio tracks. So that if I run the command on a show that updated the episodes with English and German audio it will then compare chrunchyroll = local file and compare the audio tracks available and if chrunchyroll has additional audio tracks specified in the command line ... -a en-US -a de-DE -a ja-JP ... it will re-download the episode and replace it on the local storage.

Would that be possible or is there a different way to accomplish that ?

Thank you for your time.

**Comment by exus85** [comment]
I would also like to see this improvement, if possible

**Comment by Frooastside** [comment]
i think you are looking for the `--skip-existing-method` cli argument. from the readme:

> Skip existing method
> By default, already existing files are determined by their name and the download of the corresponding episode is skipped. But sometimes Crunchyroll adds dubs or subs to an already existing episode and these changes aren't recognized and --skip-existing just skips it. This behavior can be changed by the --skip-existing-method flag. Valid options are audio and subtitle (if the file already exists but the audio/subtitle are less from what should be downloaded, the episode gets downloaded and the file overwritten).

```bash
$ crunchy-cli archive --skip-existing-method audio --skip-existing-method video https://www.crunchyroll.com/series/GY8VEQ95Y/darling-in-the-franxx
```
